multi_company_holding_invoicing/wizards/base_invoicing.py

```diff
@@ -26,50 +21,78 @@ def _get_invoice_line_data(self, data):
 
     @api.model
     def _get_accounting_value_from_product(self, data_line, product):
-        # We do not have access to the partner here so we can not
-        # play correctly the onchange
-        # We need to refactor the way to generate the line
-        # Refactor will be done in V10
-        # for now we just read the info on the product
-        # you need to set the tax by yourself
-        if self._context.get('section_group_by') == 'none':
+        company = self.env['res.company'].browse(
+            self.env.context['force_company'])
+        if self.env.context.get('agree_group_by') == 'none':
             name = product.name
         else:
             name = '%s - %s' % (
                 data_line['name'], data_line.get('client_order_ref', ''))
+        # get invoice line data from product onchange
+        invoice = self.env['account.invoice'].browse(
+            data_line.get('invoice_id'))
+        line_data = {
+            'product_id': product.id,
+            'uom_id': product.uom_id.id,
+            'name': '',
+            'company_id': company.id,
+            'partner_id': invoice.partner_id,
+            'invoice_id': data_line.get('invoice_id'),
+        }
+        line_data = self.env['account.invoice.line'].play_onchanges(
+            line_data, ['product_id'])
+        account_id = line_data.get('account_id', False)
+        if account_id:
+            account = self.env['account.account'].browse(account_id)
+        else:
```

Comment:
Not sure that we need this.
Maybe the best is simply to raise an error if the account is missing. Indeed of picking a random one